src/lib/fields_derivers_snapps/fields_derivers_snapps.ml

```diff
@@ -146,6 +162,14 @@ module Derivers = struct
     let _b = Fields_derivers_json.To_yojson.contramap ~f:contramap x obj in
     Fields_derivers_json.Of_yojson.map ~f:map x obj
 
+  let iso_record ~of_record ~to_record record_deriver obj =
```

Comment:
I would love if this actually worked the way we wanted it to, I think iso fails us somewhere here though -- but now I can't remember. I suppose we'll find out when we look at the schema

Reply:
@bkase At first I thought as well .. but then I realized that it contramap / map just have to be applied in the final step before using whatever is created. In JSON that is why I used the maps in the final to/of_json. In GraphQL the final step may be the schema creation? Or in any case for all the records I used it on it's fine because they become fields of other Schema.objs in the end

Reply:
😮 that makes sense! This implementation works fine then as long as we always use it within fields! Great!